api/src/utils/generics/base-repository.ts

```
@@ -166,33 +173,35 @@ export abstract class BaseRepository<
       await repository.postDelete(query, result);
     });
 
-    hooks?.findOneAndUpdate.pre.execute(async function () {
+    hooks.findOneAndUpdate.pre.execute(async function () {
       const query = this as Query<D, D, unknown, T, 'findOneAndUpdate'>;
       const criteria = query.getFilter();
       const updates = query.getUpdate();
-
-      await repository.preUpdate(query, criteria, updates);
-      repository.emitter.emit(
-        repository.getEventName(EHook.preUpdate),
-        criteria,
-        updates?.['$set'],
-      );
+      if (updates) {
```

If the updates is undefined, I think we should throw an error.